### PR TITLE
EN-1639 Update Entity Attribute payload element (5.0.1)

### PR DESCRIPTION
**Pull request by eugeniosant**

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

File: engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java [hunk 1]
```diff
@@ -83,14 +80,14 @@ public EntityAttributeDto(AttributeInterface src) {
             list.stream().forEach(element -> this.getElements().add(new EntityAttributeDto(element)));
         } else if (src instanceof CompositeAttribute) {
             Map<String, AttributeInterface> map = ((CompositeAttribute) src).getAttributeMap();
-            map.keySet().stream().forEach(key -> this.getElements().add(new EntityAttributeDto(map.get(key))));
+            map.keySet().stream().forEach(key -> this.getCompositeelements().add(new EntityAttributeDto(map.get(key))));
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java]: Fix casing on method call. getCompositeElements (missing capital e)

File: engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java [hunk 2]
```diff
@@ -83,14 +80,14 @@ public EntityAttributeDto(AttributeInterface src) {
...
         } else if (src instanceof ListAttribute) {
             Map<String, List<AttributeInterface>> map = ((ListAttribute) src).getAttributeListMap();
             map.keySet().stream().forEach(key -> {
                 List<EntityAttributeDto> dtos = new ArrayList<>();
                 List<AttributeInterface> list = map.get(key);
                 list.stream().forEach(element -> dtos.add(new EntityAttributeDto(element)));
-                this.getListElements().put(key, dtos);
+                this.getListelements().put(key, dtos);
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java]: Fix method name

File: engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityDto.java [hunk 3]
```diff
@@ -73,7 +73,7 @@ public void fillEntity(IApsEntity prototype, ICategoryManager categoryManager, B
         prototype.setDescription(this.getDescription());
         prototype.setMainGroup(this.getMainGroup());
         if (null != this.getGroups()) {
-            this.getGroups().stream().forEach(i -> prototype.addGroup(typeCode));
+            this.getGroups().stream().forEach(groupCode -> prototype.addGroup(groupCode));
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityDto.java]: Does this need to be streamed? prototype.getGroups().addAll(this.getGroups());

Or expose an addList method.

File: engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java (hunk 1 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java]: Done

File: engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java (hunk 2 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityAttributeDto.java]: Done

File: engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityDto.java (hunk 3 above)
Comment (eugeniosant) [reply engine/src/main/java/org/entando/entando/aps/system/services/entity/model/EntityDto.java]: Done

**Comment by entando-service** [comment]
All is well

**Pull request merged**